ntoskrnl/mm/ARM3/sysldr.c

```diff
@@ -2964,7 +2964,52 @@ MmLoadSystemImage(IN PUNICODE_STRING FileName,
     PrefixName = *FileName;
 
     /* Check if we have a prefix */
-    if (NamePrefix) DPRINT1("Prefixed images are not yet supported!\n");
+    if (NamePrefix)
+    {
+        /* Check if "directory + prefix" is too long for the string */
+        Status = RtlUShortAdd(NamePrefix->Length,
+                              BaseDirectory.Length,
```

Comment:
Putting in the order the elements show up in the string (directory, then nameprefix, then basename)
```suggestion
        Status = RtlUShortAdd(BaseDirectory.Length,
                              NamePrefix->Length,
```